packages/swapper/src/swappers/osmosis/OsmosisSwapper.ts

```diff
@@ -0,0 +1,410 @@
+    }
+  }
+
+  async buildTrade(args: BuildTradeInput): Promise<Trade<ChainId>> {
+    const { sellAsset, buyAsset, sellAmount } = args
+
+    if (!sellAmount) {
+      throw new Error('sellAmount is required')
+    }
+
+    const { rate, buyAmount } = await getRateInfo(
+      sellAsset.symbol,
+      buyAsset.symbol,
+      sellAmount !== '0' ? sellAmount : '1',
+      this.deps.osmoUrl
+    )
+
+    //convert amount to base
+    let amountBaseSell: number = parseFloat(sellAmount)
+    amountBaseSell = parseInt(String(amountBaseSell))
+    const amountBaseSellString = amountBaseSell.toString()
+
+    return {
+      buyAmount,
+      buyAsset,
+      feeData: { fee, tradeFee: fee }, // TODO: and real tradeFee
+      rate,
+      receiveAddress: '',
+      sellAmount: amountBaseSellString,
+      sellAsset,
+      sellAssetAccountNumber: 0,
+      sources: [{ name: 'Osmosis', proportion: '100' }]
+    }
+  }
+
+  async getTradeQuote(input: CommonTradeInput): Promise<TradeQuote<ChainId>> {
+    const { sellAsset, buyAsset, sellAmount } = input
+    if (!sellAmount) {
+      throw new Error('sellAmount is required')
+    }
+    const { rate, buyAmount } = await getRateInfo(
+      sellAsset.symbol,
+      buyAsset.symbol,
+      sellAmount !== '0' ? sellAmount : '1',
+      this.deps.osmoUrl
+    )
+
+    const { minimum, maximum } = await this.getMinMax(input)
+
+    return {
+      buyAsset,
+      feeData: { fee, tradeFee: fee }, // TODO: give real trade fee
+      maximum,
+      minimum,
+      sellAssetAccountNumber: 0,
+      rate,
+      sellAsset,
+      sellAmount,
+      buyAmount,
+      sources: DEFAULT_SOURCE,
+      allowanceContract: 'Allowance' // TODO: what is this
+    }
+  }
+
+  async executeTrade(args: ExecuteTradeInput<ChainId>): Promise<TradeResult> {
+    const sellAsset = args.trade.sellAsset
+    const buyAsset = args.trade.buyAsset
+    let sellAmount = args.trade.sellAmount
+    const buyAmount = args.trade.buyAmount
+    const wallet = args.wallet
+
+    if (!sellAsset) throw Error('missing sellAsset')
+    if (!buyAsset) throw Error('missing buyAsset')
+    if (!sellAmount) throw Error('missing sellAmount')
+    if (!buyAmount) throw Error('missing buyAmount')
+    if (!wallet) throw Error('missing wallet')
+
+    const pair = sellAsset.symbol + '_' + buyAsset.symbol
+    const sellAssetDenom = symbolDenomMapping[sellAsset.symbol as keyof IsymbolDenomMapping]
+    const buyAssetDenom = symbolDenomMapping[buyAsset.symbol as keyof IsymbolDenomMapping]
+
+    const gas = '1350000'
+
+    const osmosisAdapter = this.deps.adapterManager.get(osmosisChainId) as
+      | osmosis.ChainAdapter
+      | undefined
+    const cosmosAdapter = this.deps.adapterManager.get(cosmosChainId) as
+      | cosmos.ChainAdapter
+      | undefined
+
+    if (cosmosAdapter && osmosisAdapter) {
+      const osmoAddress = await (wallet as OsmosisWallet).osmosisGetAddress({
+        addressNList: bip32ToAddressNList("m/44'/118'/0'/0/0")
```

Comment:
This should involve the sell asset account number that we pass in. Something like how we do in the ZrxSwapper:

    const bip44Params = adapter.buildBIP44Params({ accountNumber: Number(buyAssetAccountNumber) })
    const receiveAddress = await adapter.getAddress({ wallet, bip44Params })

Reply:
I directly copied that from zrx but the idea is these derivation paths should contain that value passed in